bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoAbstractHandler.java

```diff
@@ -216,7 +220,8 @@ public void dispose() {
             }
         }
         scheduledJobs.clear();
-        miIoScheduler = ThreadPoolManager.getScheduledPool(BINDING_ID);
+        miIoScheduler.shutdownNow();
+        miIoScheduler = scheduler;
```

Comment:
Don't set it to `scheduler` here, it would just allow further tasks to be scheduled after disposal.